File: builtins/BigInt.js
var ops = __ops__;
var DefineProp = ops.DefineProp;
var _bigIntToString = BigInt.prototype.__toString;
var _bigIntValueOf = BigInt.prototype.__valueOf;
var _bigIntAsIntN = BigInt.__asIntN;
var _bigIntAsUintN = BigInt.__asUintN;

BigInt.prototype.toString = function BigIntToString() {
  return _bigIntToString.call(this, arguments[0]);
};

BigInt.prototype.valueOf = function BigIntValueOf() {
  return _bigIntValueOf.call(this);
};

BigInt.asIntN = function BigIntAsIntN(bits, bigint) {
  return _bigIntAsIntN.call(this, bits, bigint);
};

BigInt.asUintN = function BigIntAsUintN(bits, bigint) {
  return _bigIntAsUintN.call(this, bits, bigint);
};

File: crates/quench-runtime/src/builtin_meta/bigint.rs
//! BigInt method metadata.

use crate::ops::Builtin;

pub const fn fn_name(b: Builtin) -> Option<&'static str> {
    match b {
        Builtin::BigIntAsIntN => Some("BigInt.asIntN"),
        Builtin::BigIntAsUintN => Some("BigInt.asUintN"),
        Builtin::BigIntToString => Some("BigInt.prototype.toString"),
        Builtin::BigIntToLocaleString => Some("BigInt.prototype.toLocaleString"),
        Builtin::BigIntValueOf => Some("BigInt.prototype.valueOf"),
        _ => None,
    }
}

pub const fn fn_len(b: Builtin) -> Option<f64> {
    match b {
        Builtin::BigIntAsIntN | Builtin::BigIntAsUintN => Some(2.0),
        Builtin::BigIntToString | Builtin::BigIntToLocaleString | Builtin::BigIntValueOf => {
            Some(0.0)
        }
        _ => None,
    }
}

pub const fn short_name(b: Builtin) -> Option<&'static str> {
    match b {
        Builtin::BigIntAsIntN => Some("asIntN"),
        Builtin::BigIntAsUintN => Some("asUintN"),
        Builtin::BigIntToString => Some("toString"),
        Builtin::BigIntToLocaleString => Some("toLocaleString"),
        Builtin::BigIntValueOf => Some("valueOf"),
        _ => None,
    }
}
